### PR TITLE
Prefer Cucumber Expressions over Regular Expressions for Java 10 minute tutorial

### LINKED ISSUE #375: Use Cucumber Expressions rather than Regular Expressions
#### Background

The 10 minute tutorial still uses Regular Expressions. Other pages possibly too.

#### Solution

We should use Cucumber Expressions everywhere (except when we explain regular expressions explicitly).

### PR DESCRIPTION
**Pull request by Riggs333**
Closes #375 

This PR also fixes a minor bug in the test path for the Java test class that we noticed during the walkthrough of the tutorial.

Compiled during a mob session with @splattael and @ryanzidago at [Hackergarten Dortmund, Germany](https://www.meetup.com/de-DE/Softwerkskammer-Ruhrgebiet/events/259817215/). :tada:

### CODE REVIEW

File: content/docs/guides/10-minute-tutorial.md [hunk 1]
```diff
@@ -879,6 +885,7 @@ package hellocucumber;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.When;
 import cucumber.api.java.en.Then;
+
```
Comment (Riggs333) [inline content/docs/guides/10-minute-tutorial.md:888]: This new line was generated by the maven archetype command. :man_shrugging: 
Hence the line number changes in the stacktraces.

File: content/docs/guides/10-minute-tutorial.md [hunk 2]
```diff
@@ -1234,36 +1241,41 @@ When we run this test, it will fail.
 
 {{% block "java" %}}
 ```shell
+-------------------------------------------------------
```
Comment (Riggs333) [inline content/docs/guides/10-minute-tutorial.md:1244]: This was missing in previous outputs. :thinking: 
Added here to be consistent with the other console output snippets. :muscle:

File: content/docs/guides/10-minute-tutorial.md [hunk 3]
```diff
@@ -1413,18 +1425,18 @@ Running hellocucumber.RunCucumberTest
...
-    Then I should be told "TGIF"       # Stepdefs.i_should_be_told(String)
-
   Scenario: Sunday isn't Friday        # hellocucumber/is_it_friday_yet.feature:4
     Given today is Sunday              # Stepdefs.today_is_Sunday()
     When I ask whether it's Friday yet # Stepdefs.i_ask_whether_it_s_Friday_yet()
     Then I should be told "Nope"       # Stepdefs.i_should_be_told(String)
 
-2 scenarios (2 passed)
-6 steps (6 passed)
+  Scenario: Friday is Friday           # hellocucumber/is_it_friday_yet.feature:9
```
Comment (Riggs333) [inline content/docs/guides/10-minute-tutorial.md:1433]: Not sure why the order of the scenarios has changed. But the new order here makes sense. :+1:

File: content/docs/guides/10-minute-tutorial.md [hunk 4]
```diff
@@ -1489,29 +1501,30 @@ package hellocucumber;
 import cucumber.api.java.en.Given;
 import cucumber.api.java.en.When;
 import cucumber.api.java.en.Then;
+
 import static org.junit.Assert.*;
 
 class IsItFriday {
     static String isItFriday(String today) {
-	return "Friday".equals(today) ? "TGIF" : "Nope";
+        return "Friday".equals(today) ? "TGIF" : "Nope";
```
Comment (Riggs333) [inline content/docs/guides/10-minute-tutorial.md:1509]: Tabs vs. spaces :sweat_smile:

**Review by Riggs333 — COMMENTED** [review]
We left some annotations for background information for reviewer.

**Comment by Riggs333** [comment]
This seems to also fix the issue #237. :100:

File: content/docs/guides/10-minute-tutorial.md (hunk 1 above)
Comment (mpkorstanje) [reply content/docs/guides/10-minute-tutorial.md:888]: Thats okay. IIRC IDEA auto formats imports to separate static and non static imports.

File: content/docs/guides/10-minute-tutorial.md (hunk 3 above)
Comment (mpkorstanje) [reply content/docs/guides/10-minute-tutorial.md:1433]: We started ordering events in v4.

File: content/docs/guides/10-minute-tutorial.md (hunk 4 above)
Comment (mpkorstanje) [reply content/docs/guides/10-minute-tutorial.md:1509]: Yeah. :laughing:

**Review by mpkorstanje — APPROVED** [review]
Thanks! Happy Hacktober!

**Comment by mpkorstanje** [comment]
> This seems to also fix the issue #237.

Unfortunately it doesn't update the version. But that is okay. This is an improvement!

**Pull request merged**